scripts/utils/sign_and_send.js

```diff
@@ -73,7 +67,44 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     console.log("Transaction awaiting execution in the interface", interfaceLink)
   }
 
+  /**
+   * Checks that a transaction was already proposed to the gnosis-safe UI
+   *
+   * @param {Address} masterSafe Address of the master safe owning the brackets
+   * @param {Transaction} transaction The transaction to be signed and sent
+   * @param {string} network either rinkeby or mainnet
+   * @param {number} [nonce=null] specified transaction index. Will fetch correct value if not specified.
+   */
+  const checkTransactionExistenceOnSafeServer = async function (masterSafe, transaction, network, nonce = null) {
+    const safeTxGas = await estimateGas(masterSafe, transaction)
+    const transactionHash = await masterSafe.getTransactionHash(
+      transaction.to,
+      transaction.value,
+      transaction.data,
+      transaction.operation,
+      safeTxGas,
+      baseGas,
+      0,
+      ZERO_ADDRESS,
+      ZERO_ADDRESS,
+      nonce
+    )
+    console.log(`The reproduced transaction has the gnosis-safe hash ${transactionHash} and nonce ${nonce}`)
```

Comment:
What is a "gnosis-safe hash"? Can't it just be a hash? I don't think we need/should explicitly say "reproduced" here since this function is independent of the fact that the hash was reproduced for our use case.

However, I can imagine one might want to see the term reproduced if they are indeed reproducing the transaction. This being said, I can suggest that we simplify this function by removing the use case specific logs and have it return a boolean. Something like 

```py
def transactionExisitsOnSafeServer(...args) -> bool:
```

Then bubble the use case specific logs out into the script where this is called instead. This will make the function more robust for other purposes. 

It would also make me happy to get the word "check" out of the function name.

Reply:
A "Gnosis Safe hash" is the hash internally used by a Gnosis Safe to identify transactions. I think it's reasonable to remark the difference from the much more common "transaction hash", which is used to identify an Ethereum transaction. 

I really like the idea of having this function return a boolean and moving out the logs.